rpcclient/local.go

```diff
@@ -120,6 +120,12 @@ func (l *Local) BroadcastTxCommit(ctx context.Context, tx types.Tx) (*ctypes.Res
 		}, nil
 	}
 
+	// broadcast tx
+	err = l.node.P2P.GossipTx(ctx, tx)
+	if err != nil {
+		return nil, fmt.Errorf("tx added to local mempool but failure to broadcast: %w", err)
+	}
+
```

Comment:
I still think it's weird that the p2p gossiping is now triggered in rpcclient but yeah let's fix this in a followup PR.